### ISSUE TITLE
Safe option not working

### ISSUE THREAD

**Issue by sm2017** [dex4er/js-sliding-window-rate-limiter#22]
I have problem with safe option

```js
const SlidingWindowRateLimiter = require("sliding-window-rate-limiter").default
const Redis = require("ioredis");

const limiter = SlidingWindowRateLimiter.createLimiter({
  interval: 60000,
  redis: new Redis({
    host: "127.0.0.1",
    retryStrategy: _times => 1000,
    maxRetriesPerRequest: 1,
  }),
  safe: true,
})

try{
 await limiter.reserve('key', 1)
}catch(e){
 console.log('it must never executed , but executed');
}


// it works now
await limiter.reserve('key', 1)
```

but when I comment this line it works
https://github.com/dex4er/js-sliding-window-rate-limiter/blob/master/src/safe-redis-sliding-window-rate-limiter.ts#L95

**Comment by dex4er** [comment]
It is because limiter itself generated `error` event that is very special in NodeJS: https://nodejs.org/api/events.html#events_error_events

You can add some listener like:

```js
limiter.on('error', () => {/* noop */})
```

And now I think I'm going to add some empty listener to the limiter so it won't be necessary to add it again outside library.

**Comment by dex4er** [comment]
Closed in v4.1.0
